File: 20221013_Thursday/20221013.go
// Leetcode: 237. Delete Node in a Linked List.
// https://leetcode.com/problems/delete-node-in-a-linked-list/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 6 ms, faster than 41.86% of Go online submissions for Delete Node in a Linked List.
// Memory Usage: 2.9 MB, less than 67.73% of Go online submissions for Delete Node in a Linked List.
// 2022.10.13 Daily Challenge.
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteNode(node *ListNode) {
  var prev *ListNode
  prev = nil
  for ; node.Next != nil ; {
    node.Val = node.Next.Val
    prev = node
    node = node.Next
  }
  prev.Next = nil
}

File: 20221013_Thursday/20221013.rb
# Leetcode: 237. Delete Node in a Linked List.
# https://leetcode.com/problems/delete-node-in-a-linked-list/
# = = = = = = = = = = = = = =
# Accepted.
# Thanks God, Jesus Christ!
# = = = = = = = = = = = = = =
# Runtime: 183 ms, faster than 8.00% of Ruby online submissions for Delete Node in a Linked List.
# Memory Usage: 210.9 MB, less than 40.00% of Ruby online submissions for Delete Node in a Linked List.
# 2022.10.13 Daily Challenge.
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} node
# @return {Void} Do not return anything, modify node in-place instead.
def delete_node(node)
  current = node
  previous = nil
  while current.next
    current.val = current.next.val
    previous = current
    current = current.next
  end
  previous.next = nil
end
